api/v1alpha1/reference_types.go

```diff
@@ -0,0 +1,57 @@
+// SPDX-FileCopyrightText: 2022 SAP SE or an SAP affiliate company and Open Component Model contributors.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package v1alpha1
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/fluxcd/pkg/apis/meta"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	ocmmetav1 "github.com/open-component-model/ocm/pkg/contexts/ocm/compdesc/meta/v1"
+	v3alpha1 "github.com/open-component-model/ocm/pkg/contexts/ocm/compdesc/versions/ocm.software/v3alpha1"
+)
+
+// ObjectReference defines a resource which may be accessed via a snapshot or component version
+// +kubebuilder:validation:MinProperties=1
+type ObjectReference struct {
+	meta.NamespacedObjectKindReference `json:",inline"`
+
+	// +optional
+	ResourceRef *ResourceReference `json:"resourceRef,omitempty"`
+}
+
+type ResourceReference struct {
+	// +required
+	v3alpha1.ElementMeta `json:",inline"`
+
+	// +optional
+	ReferencePath []ocmmetav1.Identity `json:"referencePath,omitempty"`
+}
+
+func (o *ObjectReference) GetNamespacedName() string {
+	return fmt.Sprintf("%s/%s", o.Namespace, o.Name)
+}
+
+func (o *ObjectReference) GetObjectKey() client.ObjectKey {
+	return client.ObjectKey{Namespace: o.Namespace, Name: o.Name}
+}
+
+func (o *ObjectReference) GetGVR() schema.GroupVersionResource {
```

Comment:
Doing this I'm wondering. Would it be that bad to make the Source a `Group` object so the YAML would define these fields for use instead?